harness/Makefile

```
@@ -102,6 +102,35 @@ observability: config
+# Spawn the `context-compaction` worker as a background process. Kept OUT of
+# `iii.worker.yaml` `dependencies:` because the upstream registry doesn't yet
+# know about it; that means `iii worker add` can't resolve it and it stays
+# absent from `EXPECTED_WORKERS` / `harness::status`. Mirrors the same
+# "optional, side-of-config" pattern used by `iii-observability`. The worker
+# is its own iii-sdk client — it connects to the engine over WS on its own
+# once started.
+#
+# Requires the engine to be up first. The PID file lives under $(PIDS_DIR)
+# alongside `engine.pid` / `web.pid` so `make stop` cleans it up uniformly.
+COMPACTION_BIN := $(WORKERS_BIN)/context-compaction
+compaction: ensure-dirs
```

checkmake (0.3.2)

[warning] Target body for "compaction" exceeds allowed length of 5 lines (14).

(maxbodylength)